lib/maths/CSeasonalTime.cc

```diff
@@ -25,8 +25,8 @@ namespace ml {
 namespace maths {
 namespace {
 // DO NOT change the existing tags if new sub-classes are added.
-const std::string DIURNAL_TIME_TAG("a");
-const std::string ARBITRARY_PERIOD_TIME_TAG("b");
+const core::TPersistenceTag DIURNAL_TIME_TAG{"a", "diurnal-time"};
```

Comment:
Underscore would be better than hyphen in a field name that might be rendered in JSON.